apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryOptions.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+namespace Google.Cloud.Storage.V1;
+
+/// <summary>
+/// Options to pass custom retry configuration for each API request
+/// </summary>
+public sealed class RetryOptions
+{
+    /// <summary>
+    /// The default configuration used for retry upon failure
+    /// </summary>
+    public static RetryOptions IdempotentRetryOptions { get; } = new RetryOptions(RetryTimings.Default, RetryPredicate.DefaultErrorCodes);
+
+    /// <summary>
+    /// This configuration ensures that the api request is not retried upon failure 
+    /// </summary>
+    public static RetryOptions Never { get; } = new RetryOptions(new RetryTimings(), RetryPredicate.FromErrorCodes());
```

Comment:
Even when we don't retry, do we want to pass a new RetryTimings object to the constructor? What happens if we pass null?

Reply:
Ok. `RetryOptions` is public. What if the user passes the `RetryTimings` as `null`?